src/commands/publish/mod.rs

```diff
@@ -17,18 +18,30 @@ use crate::commands::kv;
 use crate::commands::subdomain::Subdomain;
 use crate::http;
 use crate::settings::global_user::GlobalUser;
+
 use crate::settings::target::Target;
 use crate::terminal::{emoji, message};
 
 pub fn publish(
     user: &GlobalUser,
-    target: &Target,
+    target: &mut Target,
     push_worker: bool,
     push_bucket: bool,
 ) -> Result<(), failure::Error> {
     info!("workers_dev = {}", target.workers_dev);
 
     validate_target(target)?;
+
+    if let Some(site_config) = &target.site {
+        let site_namespace = kv::namespace::site(target, user)?;
+
+        target.add_kv_namespace(KvNamespace {
+            binding: "__STATIC_CONTENT".to_string(),
+            id: site_namespace.id,
+            bucket: Some(site_config.bucket.to_owned()),
+        });
+    }
```

Comment:
considered stuffing all of this logic in the kv_namespaces method on target, but because there is an api call, we need the global user, and it feels wrong to add that dependency to target. since we don't re-use clients in this function, there is also a world in which we initialize the user and the client inside the kv::namespace::site function, and I think that might be OK, but I think I want to wait until I implement this for preview.

Reply:
^ this (putting user and client logic into kv::namespace::site) sounds appropriate!